packages/modules/payment/src/services/payment-module.ts

```diff
@@ -932,7 +939,45 @@ export default class PaymentModuleService
         filters.context
       )
 
-    return [paymentMethods, paymentMethods.length]
+    const normalizedResponse = paymentMethods.map((item) => ({
+      id: item.id,
+      data: item.data,
+      provider_id: filters.provider_id,
+    }))
+
+    return [normalizedResponse, paymentMethods.length]
+  }
+
+  // @ts-ignore
+  createPaymentMethods(
+    data: CreatePaymentCollectionDTO,
+    sharedContext?: Context
+  ): Promise<PaymentCollectionDTO>
+
+  createPaymentMethods(
+    data: CreatePaymentMethodDTO[],
+    sharedContext?: Context
+  ): Promise<PaymentMethodDTO[]>
+  @InjectManager()
+  async createPaymentMethods(
+    data: CreatePaymentMethodDTO | CreatePaymentMethodDTO[],
+    @MedusaContext() sharedContext?: Context
+  ): Promise<PaymentMethodDTO | PaymentMethodDTO[]> {
+    const input = Array.isArray(data) ? data : [data]
+
+    const result = await Promise.all(
```

Comment:
suggestion: I believe we should use promiseAll instead. also, if one of the payment method fail, what should we do? since the other one might have succeed?

Reply:
with promiseAll, you can get back all the one that have failed and probably handle that error in some ways? wdyt?

Reply:
I implemented the bulk support simply to match the interface with other methods, but I doubt it will often be used like that. In the case it is used like that, I think if one fails the method should throw, since you intentionally decided to run then in bulk. But this can be changed later on if that's not true.

Reply:
yes I do agree, but here if it fails some might still have succeeded but you have not information about what happen, which have failed etc? don't you think using promiseAll (our own util) would be better and then you have all the one that have failed and you can rethrow a proper message saying that X and Y have failed because of... ?

Reply:
we can still do that later it is not blocking but I am just wondering

Reply:
Ah I am not challenging using `promiseAll`, I will definitely use that. Just saying that what is expected to happen might depend on the use-case :)

Reply:
Pushed a new commit with the changes+